flink-table/flink-table-common/src/test/java/org/apache/flink/table/factories/TestDynamicTableFactory.java

```diff
@@ -50,10 +50,16 @@
     public static final String IDENTIFIER = "test-connector";
 
     public static final ConfigOption<String> TARGET =
-            ConfigOptions.key("target").stringType().noDefaultValue();
+            ConfigOptions.key("target")
+                    .stringType()
+                    .noDefaultValue()
+                    .withDeprecatedKeys("deprecated-target");
```

Comment:
(Going on a bit of a tangent here, but we should eventually get to a point where we can test these things in isolation rather than having a single factory which tries to implement every test(able) scenario at once. Of course this is out of scope for this PR.)

Reply:
I agree. It should be easier to hook into the discovery process.